racun-service/src/main/java/rs/edu/raf/banka/racun/controller/UgovorController.java

```diff
@@ -62,60 +76,48 @@ public ResponseEntity<?> getAllKompanija(@RequestHeader("Authorization") String
 
     @GetMapping(value = "/company/{kompanijaId}/finalized", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<?> getAllKompanijaFinalized(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
-        try
-        {
-            var ugovori = ugovorService.getAllByCompanyFinalized(kompanijaId);
+        try {
+            var ugovori = ugovorService.getAllByCompanyFinalized(kompanijaId, token);
             return ResponseEntity.ok(ugovori);
-        }
-        catch (Exception ex)
-        {
+        } catch (Exception ex) {
             return ResponseEntity.badRequest().body(ex.getMessage());
         }
     }
 
     @GetMapping(value = "/company/{kompanijaId}/draft", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<?> getAllKompanijaDraft(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
-        try
-        {
-            var ugovori = ugovorService.getAllByCompanyDraft(kompanijaId);
+        try {
+            var ugovori = ugovorService.getAllByCompanyDraft(kompanijaId, token);
             return ResponseEntity.ok(ugovori);
-        }
-        catch (Exception ex)
-        {
+        } catch (Exception ex) {
             return ResponseEntity.badRequest().body(ex.getMessage());
         }
     }
 
     @PostMapping(value = "/", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<?> createUgovor(@RequestHeader("Authorization") String token, @RequestBody UgovorCreateRequest request) {
-        try
-        {
-            var ugovor = ugovorService.createUgovor(request);
+        try {
+            var ugovor = ugovorService.createUgovor(request, token);
```

Comment:
Ovde ne treba token jer kreiranje ugovora mogu da vrse i agenti i supervizori.

Reply:
Token je potreban da bi se sacuvao agentId koj je kreirao ugovor

Reply:
Da, video sam sad. Ono sto bih izmenio to je da u modelu bude userId umesto agentId i onda treba da dodas u servisu prilikom pravljenja ugovora serUserId(dohvatis id preko tokena). Takodje svuda gde pise agentId treba preimenovati na userId.